src/System Application/App/Feature Key/src/FeatureManagementImpl.Codeunit.al

```diff
@@ -106,8 +107,10 @@ codeunit 2610 "Feature Management Impl."
             end;
         // If the table extension is not in sync during upgrade then Get() always returns False,
         // so the following insert will fail if the record does exist.
-        if AllowInsert then
+        if AllowInsert then begin
             if FeatureDataUpdateStatus.Insert() then;
+            Session.LogAuditMessage(StrSubstNo(FeatureKeyStatusChangedLbl, FeatureDataUpdateStatus."Feature Key", FeatureDataUpdateStatus."Feature Status", UserSecurityId()), SecurityOperationResult::Success, AuditCategory::ApplicationManagement, 4, 0);
+        end;
```

Comment:
You can also add else statement and add 2 at the end to indicate that it has failed.

Reply:
Will skip negative case, as it is used in upgrade too.